pandas/core/nanops.py

```diff
@@ -474,8 +474,7 @@ def nanargmax(values, axis=None, skipna=True):
     """
     Returns -1 in the NA case
     """
-    values, mask, dtype, _ = _get_values(values, skipna, fill_value_typ='-inf',
-                                         isfinite=True)
```

Comment:
I would actually be ok with NOT allowing this for strings at all. IOW you can use the ``@disallow`` decorator. It is a slight break with numpy, but it doesn't work now anyhow. I am not sure why allowing this is a good thing when it has different py2/3 behavior (and it doesn't work on mixed things).